ras_backstage/authentication/token_decoder.py

```diff
@@ -0,0 +1,33 @@
+import logging
+
+from flask import abort
+import jwt
+from jwt import DecodeError
+
+from ras_backstage.authentication.uaa import get_uaa_public_key
+
+
+logger = logging.getLogger(__name__)
+
+
+def decode_access_token(access_token, verify=True):
+
+    try:
+        decoded_jwt = jwt.decode(
+            access_token,
+            verify=verify,
+            algorithms=None,
+            key=get_uaa_public_key(),
+            audience='ras_backstage',
+            leeway=10,
+        )
+        return decoded_jwt
+    except DecodeError:
+        logger.error(f"Unable to decode token {access_token} - confirm the UAA public key is correct")
+        abort(500)
+
+
+def get_user_id(access_token, verify=True):
+    decoded_jwt = decode_access_token(access_token, verify)
+    if decoded_jwt:
```

Comment:
Will decoded_jwt ever evaluate to None? it looks like it will either succesfully decode or abort(500).

If not, this could be ```return decode_access_token(access_token, verify).get('user_id')```

In which case this function is obsolete.

Reply:
Good point, think it was from when I had decode_access_token return None on failure. I'll change